src/controllers/project.js

```diff
@@ -0,0 +1,131 @@
+const Project = require('../models/project');
+
+const getAllProjects = async (req, res, next) => {
+  const projects = await Project.find({});
+  try {
+    if (!projects) {
```

Comment:
this whole function is not written correctly there's too much repetition with the error, please do the following changes
1- move Project.find inside the try block
2- catch the error and return it in the response with the error status code and message example code

```
const getAllBlogPosts = async (_, res) => {
  try {
    // Simple find query to fetch all blog posts
    const blogPosts = await BlogPostModel.find();
    res.json(blogPosts);
  } catch (err) {
    res.status(422).json({ message: err.message });
  }
};
```
please apply similar changes to your all other functions cause I see the same incorrect pattern happening.

Reply:
After your do these changes ask for another review.